src/sentry/integrations/utils/sync.py

```diff
@@ -120,5 +134,10 @@ def sync_group_assignee_outbound(group: Group, user_id: int | None, assign: bool
 
     for external_issue_id in external_issue_ids:
         sync_assignee_outbound.apply_async(
-            kwargs={"external_issue_id": external_issue_id, "user_id": user_id, "assign": assign}
+            kwargs={
+                "external_issue_id": external_issue_id,
+                "user_id": user_id,
+                "assign": assign,
+                "assignment_source_json": assignment_source.json() if assignment_source else None,
```

Comment:
out of curiousity, what's the difference between dumping this as a json and then reconstructing the class from the json vs dumping it as a dictionary? is it because one of the values (`AssignmentSourceType`) is another class?

Reply:
Dumping as a dictionary would work as well. We should avoid passing the dataclass instance as that relies on celery using pickle for task parameters which we'd like to phase out.

Reply:
Removed the nested enum type and converted this to use dictionaries instead since that's a bit cleaner. 

>  is it because one of the values (AssignmentSourceType) is another class?
Yep! Having the nested class was preventing me from doing this before.